docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,529 @@
+With the increased blast radius of the policies comes great responsibility, 
+we don't want small mistake in the policy to break traffic on the other side of the world.
+A few things could be done to protect users from impactful mistakes:
+* Enhanced validation. If policy is created in `backend-ns` of the `zone-1` and the top-level targetRef is `Mesh`, 
+then `to[].targetRef` should have both `k8s.kuma.io/namespace: backend-ns` and `kuma.io/zone: zone-1`. 
+* Ability to opt-in or opt-out from syncing the policy to other zones.
+
+#### Enhanced validation
+
+Since applying a policy to the custom namespace of a zone has a potential to change envoy configs of all pods in the mesh including pods in other zones,
+we need to restrict "to" section, so it affect only the requests going to the custom namespace.
+For example, the following policy when applied to `backend-ns` is too intrusive toward the backend consumers:
+
+```yaml
+kind: MeshTimeout
+metadata:
+  namespace: backend-ns
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          kuma.io/display-name: redis
+          k8s.kuma.io/namespace: redis-ns
+      default:
+        connectionTimeout: 5s # why does the Backend Service Owner try to change the way clients use the 'redis' service?
+```
+
+Such policy should not be allowed, if Backend Service Owner needs to change the way clients use `redis`
+they have to put the policy into `kuma-system` namespace by checking with Mesh Operator first.
+
+The validation rules for namespace-scoped policy are the following:
+* to-policy when placed into the custom namespace must have the namespace and zone either in the top-level targetRef or in each to-item
+* from-policy when placed into the custom namespace must always have the namespace and zone in the top-level targetRef
+
+For example, allowed policies are:
+```yaml
+metadata:
+  namespace: backend-ns
+  labels:
+    kuma.io/origin: zone
+    kuma.io/zone: producer-zone
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          kuma.io/display-name: backend
+          k8s.kuma.io/namespace: backend-ns
+          kuma.io/zone: producer-zone
+---
+metadata:
+  namespace: backend-ns
+  labels:
+    kuma.io/origin: zone
+    kuma.io/zone: producer-zone
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      k8s.kuma.io/namespace: backend-ns
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          kuma.io/display-name: redis
+          kuma.io/zone: producer-zone
+---
+metadata:
+  namespace: backend-ns
+  labels:
+    kuma.io/origin: zone
+    kuma.io/zone: producer-zone
+spec:
+  targetRef:
+    kind: MeshSubset
+    tags:
+      k8s.kuma.io/namespace: backend-ns
+      kuma.io/zone: producer-zone
+  to:
+    - targetRef:
+        kind: MeshService
+        name: redis
+        namespace: redis-ns
+    - targetRef:
+        kind: Mesh
```

Comment:
I just get confused with the example, it looks like a cross namespace control? 
So, can we define our top TargetRef `MeshSubset` with another NS `front-end` if our policy is `backend-ns`?

Actually, I feel it's pretty complex in the cross-namespace use cases :(

Reply:
OK, I guess it's normal that we have cross namespace application communications use cases. 
What about in these cases, the policy must use `kuma-system` namespace?

Reply:
OK, I just get understand with this.
Maybe we should also paste the link here: https://gateway-api.sigs.k8s.io/mesh/#gateway-api-for-service-mesh

For those people who are unfamiliar with the context

Reply:
I added the link to GAMMA, thanks